synapse/storage/databases/main/events_worker.py

```diff
@@ -600,7 +600,10 @@ async def _get_events_from_cache_or_db(
         Returns:
             map from event id to result
         """
-        event_entry_map = await self._get_events_from_cache(
+        # Shortcut: check if we have any events in the *in memory* cache - this function
+        # may be called repeatedly for the same event so at this point we cannot reach
+        # out to any external cache for performance reasons.
```

Comment:
add a note to say we check external caches later, in `get_missing_events_from_cache_or_db` ?